pennylane/transforms/core/transform.py

```diff
@@ -217,6 +222,50 @@ def qnode_circuit(a):
           and a transformed :class:`pennylane.devices.Device` is returned. The transform is added
           to the end of the device program and will be last in the overall transform program.
 
+    .. details::
+        :title: Transforms with Catalyst and MLIR
+
+        Transforms can be matched with an associated MLIR or XDSL pass via the ``pass_name`` argument.
+        These transforms will be lowered into the MLIR and applied as part of the lower level compilation.
+
```

Comment:
```suggestion

        For example, consider the :func:`catalyst.passes.disentangle_cnot` pass:

```

This might be a bit of an easier example to follow; `removed-chained-self-inverse` isn't very forward facing 🤔

Reply:
Here I'm wanting showing creating a transform via the `pass_name`, rather than using an already existing transform, since this docstring is about how to create transforms. We can document elsewhere abou thow to use them.

I'll make a note that `"remove-chained-self-inverse"` is the pass name for `cancel_inverses`.

Reply:
> I'll make a note that "remove-chained-self-inverse" is the pass name for cancel_inverses.

Just a note that this is no longer true. The pass name for `cancel_inverses` is now `cancel-inverses`